include/clang/AST/Expr.h

```diff
@@ -2962,6 +2962,77 @@ class CStyleCastExpr final
   friend class CastExpr;
 };
 
+class BoundsCastExpr final
+    : public ExplicitCastExpr,
+      private llvm::TrailingObjects<BoundsCastExpr, CXXBaseSpecifier *> {
+  SourceLocation LPLoc;
+  SourceLocation RPLoc;
+  Expr *BaseExpr = nullptr;
+  Expr *CountExpr = nullptr;
```

Comment:
I think this means a lot. When I though I have to choices to implement this castings. One is build bounds expression after the parsing and the other is build bounds expression in CheckFunctionBodyBoundsDecls. and depending on how it is implemented, It would be decided if I make new Bounds Cast Class or just add bounds expression pointer in CStyleCast. Currently I implemented the first way, and that's why there are separate fields for CountExpr and RangeExpr. May be I should have chosen second option. I will change the code and update this pull request.